components/script/dom/trustedtypepolicyfactory.rs

```diff
@@ -135,24 +237,79 @@ impl TrustedTypePolicyFactoryMethods<crate::DomTypeHolder> for TrustedTypePolicy
+        let element_namespace = match element_namespace {
+            Some(namespace) if !namespace.is_empty() => Namespace::from(namespace),
+            _ => ns!(html),
+        };
+        // Step 4: If attrNs is an empty string, set attrNs to null.
+        let attribute_namespace = match attribute_namespace {
+            Some(namespace) if !namespace.is_empty() => Some(Namespace::from(namespace)),
+            _ => None,
+        };
+        // Step 5: Let interface be the element interface for localName and elementNs.
+        let interface = QualName::new(None, element_namespace, LocalName::from(local_name));
+
+        // Step 6, 7, 8 & 9: Let expectedType be null.
+        // Set attributeData to the result of Get Trusted Type data for attribute algorithm,
+        // with the following arguments: interface as element, attribute, attrNs
+        // If attributeData is not null, then set expectedType to the interface’s name of the value
+        // of the fourth member of attributeData. Return expectedType
+        TrustedTypePolicyFactory::get_trusted_type_data_for_attribute(
+            interface,
+            attribute,
+            attribute_namespace,
+        )
     }
     /// <https://www.w3.org/TR/trusted-types/#dom-trustedtypepolicyfactory-getpropertytype>
     fn GetPropertyType(
         &self,
-        _: DOMString,
-        _: DOMString,
-        _: Option<DOMString>,
+        tag_name: DOMString,
+        property: DOMString,
+        element_namespace: Option<DOMString>,
     ) -> Option<DOMString> {
-        // TODO(36258): implement algorithm
-        Some(DOMString::from("".to_string()))
+        // Step 1: Set localName to tagName in ASCII lowercase.
+        let local_name = tag_name.to_ascii_lowercase();
+        // Step 2: If elementNs is null or an empty string, set elementNs to HTML namespace.
+        let element_namespace = match element_namespace {
+            Some(namespace) if !namespace.is_empty() => Namespace::from(namespace),
+            Some(_) | None => ns!(html),
+        };
+        // Step 3: Let interface be the element interface for localName and elementNs.
+        let interface = QualName::new(None, element_namespace, LocalName::from(local_name));
+        let property = property.str();
+
+        // Step 5: Find the row in the following table, where the first column is "*" or interface’s name,
+        // and property is in the second column. If a matching row is found, set expectedType to
+        // the interface’s name of the value of the third column.
+        for (element, prop_name, value) in &PROPERTY_TO_TRUSTED_TYPE_LOOKUP_TABLE {
+            if property == *prop_name &&
+                match element {
+                    Some((elem_ns, elem_local)) => {
+                        interface.ns == *elem_ns && interface.local == *elem_local
+                    },
+                    None => true,
+                }
```

Comment:
You can make this if statement a lot more concise by using [`Option::is_none_or`](https://doc.rust-lang.org/std/option/enum.Option.html#method.is_none_or).

I'm just saying this for future reference, as I explained in the comment above I would prefer if this this whole for loop became a chain of if statements instead.